viadot/sources/vid_club.py

```diff
@@ -1,42 +1,42 @@
 import json
 import os
 import urllib
-from datetime import datetime
-from typing import Any, Dict, List, Literal
-
+from datetime import datetime, date, timedelta
+from typing import Any, Dict, List, Literal, Tuple
 import pandas as pd
 
-from ..config import local_config
+from prefect.utilities import logging
+
 from ..exceptions import CredentialError, ValidationError
 from ..utils import handle_api_response
 from .base import Source
 
+logger = logging.get_logger()
+
 
 class VidClub(Source):
     """
     A class implementing the Vid Club API.
 
     Documentation for this API is located at: https://evps01.envoo.net/vipapi/
     There are 4 endpoints where to get the data.
-
     """
 
     def __init__(self, *args, credentials: Dict[str, Any] = None, **kwargs):
         """
         Create an instance of VidClub.
 
         Args:
-            credentials (dict, optional): Credentials to Vid Club APIs.
+            credentials (Dict[str, Any], optional): Credentials to Vid Club APIs containing token.
                 Defaults to dictionary.
 
         Raises:
-            CredentialError: If credentials are not provided in local_config or directly as a parameter.
+            CredentialError: If credentials are not provided as a parameter.
         """
-
-        DEFAULT_CREDENTIALS = local_config.get("VIDCLUB")
-        credentials = kwargs.pop("credentials", DEFAULT_CREDENTIALS)
-        if credentials is None:
-            raise CredentialError("Missing credentials.")
+        if credentials is not None:
+            self.credentials = credentials
+        else:
+            raise CredentialError("Credentials not provided.")
```

Comment:
If you are expecting credentials to be always provided in order to initialize class object maybe it could be better to not assigning value 'None' in __init__ to argument credentials and then put argument before *args with comment in docstring that this parameter is required.

Reply:
done